src/main/java/net/dv8tion/jda/internal/requests/restaction/pagination/EntitlementPaginationActionImpl.java

```diff
@@ -0,0 +1,197 @@
+import net.dv8tion.jda.api.requests.restaction.pagination.EntitlementPaginationAction;
+import net.dv8tion.jda.api.requests.restaction.pagination.PaginationAction;
+import net.dv8tion.jda.api.utils.MiscUtil;
+import net.dv8tion.jda.api.utils.data.DataArray;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.entities.EntityBuilder;
+import net.dv8tion.jda.internal.utils.Checks;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class EntitlementPaginationActionImpl
+    extends PaginationActionImpl<Entitlement, EntitlementPaginationAction>
+    implements EntitlementPaginationAction
+{
+    protected List<String> skuIds;
+    protected long guildId;
+    protected long userId;
+    protected boolean excludeEnded;
+
+    public EntitlementPaginationActionImpl(JDA api)
+    {
+        super(api, Route.Applications.GET_ENTITLEMENTS.compile(api.getSelfUser().getApplicationId()), 1, 100, 100);
+        this.skuIds = new ArrayList<>();
+        this.guildId = 0;
+        this.userId = 0;
+    }
+
+    @Nonnull
+    @Override
+    public EntitlementPaginationAction order(@Nonnull PaginationAction.PaginationOrder order)
+    {
+        if (order == PaginationOrder.BACKWARD && lastKey == 0)
+            lastKey = Long.MAX_VALUE;
+        else if (order == PaginationOrder.FORWARD && lastKey == Long.MAX_VALUE)
+            lastKey = 0;
+        return super.order(order);
+    }
+
+    @Nonnull
+    @Override
+    public EntitlementPaginationAction user(@Nullable UserSnowflake user)
+    {
+        if (user == null)
+            userId = 0;
+        else
+            userId = user.getIdLong();
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public EntitlementPaginationAction skuIds(long... skuIds)
+    {
+        this.skuIds.clear();
+        for (long skuId : skuIds)
+            this.skuIds.add(Long.toUnsignedString(skuId));
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public EntitlementPaginationAction skuIds(@Nonnull String... skuIds)
+    {
+        Checks.noneNull(skuIds, "skuIds");
+        for (String skuId : skuIds)
+            Checks.isSnowflake(skuId, "skuId");
+
+        this.skuIds.clear();
+
+        Collections.addAll(this.skuIds, skuIds);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public EntitlementPaginationAction guild(long guildId)
+    {
+        this.guildId = guildId;
+        return this;
+    }
+
+    @NotNull
+    @Override
+    public EntitlementPaginationAction guild(@NotNull String guildId)
+    {
+        Checks.notNull(guildId, "guildId");
+        Checks.isSnowflake(guildId, "guildId");
+        this.guildId = MiscUtil.parseSnowflake(guildId);
+        return this;
+    }
+
+    @NotNull
+    @Override
+    public EntitlementPaginationAction guild(@NotNull Guild guild)
```

Comment:
Wrong annotations here, also I would just make these defaults in the interface so only one method actually changes the field in `guild(long guildId)`.